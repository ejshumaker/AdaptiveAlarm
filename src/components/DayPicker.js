import React, { PureComponent } from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
import PropTypes from "prop-types";
import styled from "styled-components";
import { GlobalStyles, Colors } from "../constants";

class DayPicker extends PureComponent {
  // possibly create getters to return the state of each day button using refs
  constructor() {
    super();
    this.state = {
      sButton: null,
      mButton: null,
      tButton: null,
      wButton: null,
      thButton: null,
      fButton: null,
      saButton: null
    };
    this.updateOnPress = this.updateOnPress.bind(this);
  }

  static propTypes = {
    description: PropTypes.string
  };

  updateOnPress(type) {
    switch (type) {
      case "S":
        if (this.state.sButton === "S") {
          this.setState({ sButton: "default" });
        } else {
          this.setState({ sButton: type });
        }
        break;
      case "M":
        if (this.state.mButton === "M") {
          this.setState({ mButton: "default" });
        } else {
          this.setState({ mButton: type });
        }
        break;
      case "T":
        if (this.state.tButton === "T") {
          this.setState({ tButton: "default" });
        } else {
          this.setState({ tButton: type });
        }
        break;
      case "W":
        if (this.state.wButton === "W") {
          this.setState({ wButton: "default" });
        } else {
          this.setState({ wButton: type });
        }
        break;
      case "Th":
        if (this.state.thButton === "Th") {
          this.setState({ thButton: "default" });
        } else {
          this.setState({ thButton: type });
        }
        break;
      case "F":
        if (this.state.fButton === "F") {
          this.setState({ fButton: "default" });
        } else {
          this.setState({ fButton: type });
        }
        break;
      case "Sa":
        if (this.state.saButton === "Sa") {
          this.setState({ saButton: "default" });
        } else {
          this.setState({ saButton: type });
        }
        break;
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={() => this.updateOnPress("S")}>
          <Circle
            style={{
              backgroundColor:
                this.state.sButton === "S" ? Colors.primary : Colors.darkGray
            }}
          >
            <Text
              style={[
                styles.text,
                {
                  color:
                    this.state.sButton === "S" ? Colors.black : Colors.white
                }
              ]}
            >
              S
            </Text>
          </Circle>
        </TouchableOpacity>

        <TouchableOpacity onPress={() => this.updateOnPress("M")}>
          <Circle
            style={{
              backgroundColor:
                this.state.mButton === "M" ? Colors.primary : Colors.darkGray
            }}
          >
            <Text
              style={[
                styles.text,
                {
                  color:
                    this.state.mButton === "M" ? Colors.black : Colors.white
                }
              ]}
            >
              M
            </Text>
          </Circle>
        </TouchableOpacity>

        <TouchableOpacity onPress={() => this.updateOnPress("T")}>
          <Circle
            style={{
              backgroundColor:
                this.state.tButton === "T" ? Colors.primary : Colors.darkGray
            }}
          >
            <Text
              style={[
                styles.text,
                {
                  color:
                    this.state.tButton === "T" ? Colors.black : Colors.white
                }
              ]}
            >
              T
            </Text>
          </Circle>
        </TouchableOpacity>

        <TouchableOpacity onPress={() => this.updateOnPress("W")}>
          <Circle
            style={{
              backgroundColor:
                this.state.wButton === "W" ? Colors.primary : Colors.darkGray
            }}
          >
            <Text
              style={[
                styles.text,
                {
                  color:
                    this.state.wButton === "W" ? Colors.black : Colors.white
                }
              ]}
            >
              W
            </Text>
          </Circle>
        </TouchableOpacity>

        <TouchableOpacity onPress={() => this.updateOnPress("Th")}>
          <Circle
            style={{
              backgroundColor:
                this.state.thButton === "Th" ? Colors.primary : Colors.darkGray
            }}
          >
            <Text
              style={[
                styles.text,
                {
                  color:
                    this.state.thButton === "Th" ? Colors.black : Colors.white
                }
              ]}
            >
              Th
            </Text>
          </Circle>
        </TouchableOpacity>

        <TouchableOpacity onPress={() => this.updateOnPress("F")}>
          <Circle
            style={{
              backgroundColor:
                this.state.fButton === "F" ? Colors.primary : Colors.darkGray
            }}
          >
            <Text
              style={[
                styles.text,
                {
                  color:
                    this.state.fButton === "F" ? Colors.black : Colors.white
                }
              ]}
            >
              F
            </Text>
          </Circle>
        </TouchableOpacity>

        <TouchableOpacity onPress={() => this.updateOnPress("Sa")}>
          <Circle
            style={{
              backgroundColor:
                this.state.saButton === "Sa" ? Colors.primary : Colors.darkGray
            }}
          >
            <Text
              style={[
                styles.text,
                {
                  color:
                    this.state.saButton === "Sa" ? Colors.black : Colors.white
                }
              ]}
            >
              Sa
            </Text>
          </Circle>
        </TouchableOpacity>
      </View>
    );
  }
}

const Circle = styled.View`
  width: 35px;
  height: 40px;
  border-radius: 8px;
  justify-content: center;
  align-items: center;
  margin: 4px;
`;

const styles = StyleSheet.create({
  text: {
    fontSize: 14,
    fontWeight: "bold"
  },
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "row"
  }
});

export default DayPicker;
