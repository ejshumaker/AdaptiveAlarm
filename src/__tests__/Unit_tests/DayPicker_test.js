/* eslint-disable no-undef */
/* eslint-disable import/first */
import 'react-native';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DayPicker from '../../components/DayPicker';

// jest.mock('react-native-svg');

Enzyme.configure({ adapter: new Adapter() });

jest.setTimeout(10000);
let wrapper = null;
const onChangeDayMock = jest.fn();
const days = {
  sun: false,
  mon: false,
  tue: false,
  wed: false,
  thu: false,
  fri: true,
  sat: false,
};


describe('DayPicker tests', () => {
  beforeEach(() => {
    wrapper = shallow(<DayPicker
      onChangeDay={onChangeDayMock}
      days={days}
    />);
    jest.clearAllMocks();
  });

  it('test daypicker matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('press monday', () => {
    wrapper.find('TouchableOpacity').at(0).simulate('press', { preventDefault() {} });
    expect(onChangeDayMock).toHaveBeenCalledTimes(1);
  });

  it('press tuesday', () => {
    wrapper.find('TouchableOpacity').at(1).simulate('press', { preventDefault() {} });
    expect(onChangeDayMock).toHaveBeenCalledTimes(1);
  });
  it('press wednesday', () => {
    wrapper.find('TouchableOpacity').at(2).simulate('press', { preventDefault() {} });
    expect(onChangeDayMock).toHaveBeenCalledTimes(1);
  });
  it('press thursday', () => {
    wrapper.find('TouchableOpacity').at(3).simulate('press', { preventDefault() {} });
    expect(onChangeDayMock).toHaveBeenCalledTimes(1);
  });
  it('press friday', () => {
    wrapper.find('TouchableOpacity').at(4).simulate('press', { preventDefault() {} });
    expect(onChangeDayMock).toHaveBeenCalledTimes(1);
  });
  it('press saturday', () => {
    wrapper.find('TouchableOpacity').at(5).simulate('press', { preventDefault() {} });
    expect(onChangeDayMock).toHaveBeenCalledTimes(1);
  });
  it('press sunday', () => {
    wrapper.find('TouchableOpacity').at(6).simulate('press', { preventDefault() {} });
    expect(onChangeDayMock).toHaveBeenCalledTimes(1);
  });

  it('no days', () => {
    const days2 = {};
    wrapper2 = shallow(<DayPicker
      onChangeDay={onChangeDayMock}
      days={days2}
    />);
    expect(wrapper2).toMatchSnapshot();
  });
  afterAll(() => {
    wrapper.unmount();
  });
});
