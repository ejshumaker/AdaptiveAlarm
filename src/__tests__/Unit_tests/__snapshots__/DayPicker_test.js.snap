// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayPicker tests no days 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "space-around",
        "marginVertical": 24,
      },
    ]
  }
>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        S
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        M
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        T
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        W
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        Th
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        F
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        Sa
      </Text>
    </Styled(View)>
  </TouchableOpacity>
</View>
`;

exports[`DayPicker tests test daypicker matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "space-around",
        "marginVertical": 24,
      },
    ]
  }
>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        S
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        M
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        T
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        W
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        Th
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#E4F96A",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#13151A",
            },
          ]
        }
      >
        F
      </Text>
    </Styled(View)>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Styled(View)
      style={
        Object {
          "backgroundColor": "#505050",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RNSMiles-XBold",
              "fontSize": 14,
            },
            Object {
              "color": "#DDDDDD",
            },
          ]
        }
      >
        Sa
      </Text>
    </Styled(View)>
  </TouchableOpacity>
</View>
`;
